README.md

```diff
@@ -3,53 +3,94 @@
+to the request. Since the decryption executor places all input files in `/vol/crypt/` all input paths in subsequent
+executors are altered to `/vol/crypt/{filename}`.
+
+<img alt="request-diagram" src="images/request.png" height="600">
+
+### Decryption
+The functionality of the decryption executor lies in [`decrypt.py`][decrypt]. This script moves all input files to a
+specified output directory (in this case, `/vol/crypt/`). If a Crypt4GH file is detected and the secret key used to
+encrypt it is provided, the executor decrypts the contents of the Crypt4GH file and places it in `/vol/crypt`.
+Subsequent executors then refer to the files in `/vol/crypt/`, not their original locations.
+
+<img alt="workflow-diagram" src="images/workflow.png" height="400">
+
+## Important Considerations
+You __should not use this middleware in untrusted environments__, as it requires transmission of secret keys and stores
+the decrypted contents of Crypt4GH files on disk. This middleware is meant to be used with a [Trusted Execution 
+Environment (TEE)][TEE], which allows for the secure transmission and storage of data. Integration with TEEs is a work
+in progress.
 
 ## Installation
+```bash
+ pip install poetry
+ poetry install
+```
 
 ## Testing
-### Prerequisites
-This middleware requires a running instance of [proTES](https://github.com/elixir-cloud-aai/proTES)
-(running on localhost:8090 by default).\
-\
-Download the required dependencies\
-`pip install requirements_dev.txt`
+### Requirements
+Tests require a running TES instance and an S3 bucket containing the input files. An instance of [Funnel][funnel] 
+and [MinIO][minio] were used in development. `TES_URL` in `tests/tasks/test_tasks.py` can be altered depending on the
+TES instance being used.
+
 ### Run Tests
-Run tests using pytest\
-`pytest tests`
+Run tests using pytest.
+```bash
+poetry run pytest tests
+```
 
 ## Contributing
-
-## Code of Conduct
-
-## Versioning
+This project is a community effort and lives off your contributions, be it in the form of bug reports, feature requests,
+discussions, ideas, fixes, or other code changes. Please read these [guidelines][guidelines] if you want to contribute. 
+And please mind the [code of conduct][code-of-conduct] for all interactions with the community.
 
 ## License
-
 This project is distributed under the [Apache License 2.0][badge-license], a
 copy of which is also available in [`LICENSE`][license].
 
 ## Contact
-
 The project is maintained by [ELIXIR Cloud & AAI][elixir-cloud-aai], a Driver
 Project of the [Global Alliance for Genomics and Health (GA4GH)][ga4gh], under
 the umbrella of the [ELIXIR][elixir] [Compute Platform][elixir-compute].
 
 - For filing bug reports, feature requests or other code-related issues, please
   make use of the project's [issue tracker](https://github.com/elixir-cloud-aai/protes-middleware-crypt4gh/issues).
```

Comment:
When rendered, the single bullet point looks a little weird. Perhaps remove it until we add additional bullet points (if ever).